### PR TITLE
task 4 completed

### PR DESCRIPTION
**Pull request by Szpytma**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -52,3 +52,51 @@ describe("GET /api/categories", () => {
...
 });
+
+describe("GET /api/reviews/:review_id", () => {
+  it("should return an object of review with specific keys", () => {
+    return request(app)
+      .get("/api/reviews/1")
+      .expect(200)
+      .then(({ body }) => {
+        const { review } = body;
+        expect(review).toHaveProperty("category");
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: It's good that you're checking the keys on the review here, but if you know you're getting back a specific review you know what the category, title, id are etc. Your test would be more robust if you checked that review 1 came back correctly

File: __tests__/app.test.js [hunk 2]
```diff
@@ -52,3 +52,51 @@ describe("GET /api/categories", () => {
...
+      });
+  });
+
+  it("should return an object with the title 'Ultimate Werewolf' from id 3", () => {
+    return request(app)
+      .get("/api/reviews/3")
+      .expect(200)
+      .then(({ body }) => {
+        const { review } = body;
+        expect(review.title).toBe("Ultimate Werewolf");
```
Comment (roisinoreilly) [inline __tests__/app.test.js:78]: what if something happened to the rest of the review data and it wasn't what you're expecting?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -52,3 +52,51 @@ describe("GET /api/categories", () => {
...
+      });
+  });
+
+  it("should return an error if no number was provided as a param", () => {
+    return request(app)
+      .get("/api/reviews/one")
+      .expect(404)
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: this should be a 400

File: models/reviews.model.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const db = require("../db/");
+
+exports.fetchReviewByID = (id) => {
+  // const reg = new RegExp("[0-9]");
```
Comment (roisinoreilly) [inline models/reviews.model.js]: remember to remove any commented out code before pushing

File: models/reviews.model.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  // if (id === "" || !id) {
+  //   return Promise.reject({ status: 404, msg: "Please provide an valid ID" });
+  // }
+  const selectAllCategories = "SELECT * FROM reviews WHERE review_id  = $1";
+  return db.query(selectAllCategories, [id]).then((review) => {
+    if (review.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Index outOfBound" });
```
Comment (roisinoreilly) [inline models/reviews.model.js:7]: nicely handled

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
a great start! I'd like you to test that all of the correct keys and their values for the review you're getting back, and make sure your 404 and 400 tests are correct. After that's sorted send us another PR :)

File: __tests__/app.test.js (hunk 1 above)
Comment (Szpytma) [reply __tests__/app.test.js]: I have changed it to check for the specific values now

**Comment by Szpytma** [comment]
@roisinoreilly I have refactored all code. I look forward to hearing from you with approval.

**Review by alexdevdavis — APPROVED** [review]
Decent refactor; well done 👍  Happy for you to proceed now
- consider sending error messages in an object on the key of `message`, rather than as raw text. Will be easier for your frontend to work more dynamically if everything comes back on the `message` or `msg` key

**Pull request merged**